include/libopencm3/pac55xx/usart.h

```diff
@@ -0,0 +1,225 @@
+ *
+ * Definitions in this file come from the PAC55XX Family User Guide Rev 1.23
+ * by Active-Semi dated November 19, 2019. TX and RX hardware buffer sizes
+ * are both 16 bytes.
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2020 Kevin Stefanik <kevin@allocor.tech>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef LIBOPENCM3_PAC55XX_USART_H_
+#define LIBOPENCM3_PAC55XX_USART_H_
+
+#include <libopencm3/pac55xx/memorymap.h>
+#include <libopencm3/cm3/common.h>
+
+/**@{*/
+
+/** @defgroup usart_registers Registers
+@{*/
+/** Receive Buffer Register RO, only bits 7:0 used */
+#define USART_RBR(usart_base)   MMIO32((usart_base) + 0x0000)
+/** Transmit Holding Register WO, only bits 7:0 used */
+#define USART_THR(usart_base)   MMIO32((usart_base) + 0x0004)
+/** Divisor Latch Register RW, default 0000 0001h, only bits 15:0 used.*/
+#define USART_DLR(usart_base)   MMIO32((usart_base) + 0x0008)
+/** Interrupt Enable Register RW, default 0000 0000h */
+#define USART_IER(usart_base)   MMIO32((usart_base) + 0x000C)
+/** Interrupt Identification Register RO, default 0000 0001h */
+#define USART_IIR(usart_base)   MMIO32((usart_base) + 0x0010)
+/** FIFO Control Register RW, default 0000 0000h */
+#define USART_FCR(usart_base)   MMIO32((usart_base) + 0x0014)
+/** Line control Register RW, default 0000 0000h */
+#define USART_LCR(usart_base)   MMIO32((usart_base) + 0x0018)
+/** Line Status Register RO, default 0000 0060h */
+#define USART_LSR(usart_base)   MMIO32((usart_base) + 0x0020)
+/** Scratch Pad Register RW, only bits 7:0 used */
+#define USART_SCR(usart_base)   MMIO32((usart_base) + 0x0028)
+/** Enhanced Mode Register RW, default 0000 000h */
+#define USART_EFR(usart_base)   MMIO32((usart_base) + 0x002C)
+/**@}*/
+
+/** @defgroup usart_ier_bits Interrupt Enable Register bits
+@{*/
+/** Enable RX line status interrupt */
+#define USART_IER_RLSIE         BIT2
+/** Enable the TX Holding Empty interrupt */
+#define USART_IER_THRIE         BIT1
+/** Enable the RX Buffer Register Interrupt */
+#define USART_IER_RBRIE         BIT0
+/**@}*/
+
+/** @defgroup usart_iir_bits Interrupt ID Register bits
+@{*/
+/** This bit is active low to indicate an interrupt is pending */
+#define USART_IIR_INTSTATUS     BIT0
+/** TX Holding Register Empty */
+#define USART_IIR_TXEMPTY       (0b0010)
```

Comment:
please use hex here.  binary literals don't arrive until c14, and are a gcc extension only until then.

Reply:
fixed